src/OpenRiaServices.Client/Test/Client.Test/Data/OperationTests.cs

```diff
@@ -54,8 +56,34 @@ protected override void InvokeCompleteAction()
             this._completeAction?.Invoke(this);
         }
     }
+
+
+    class ExceptionHandleSynchronizationContext : SynchronizationContext
+    {
+        public Exception LastException { get; set; }
+        public bool HasException { get => LastException != null; }
+
+        public override void Send(SendOrPostCallback d, object state)
+        {
+            try
+            {
+                d(state);
+            }
+            catch (Exception ex)
+            {
+                LastException = ex;
+                throw;
```

Comment:
I think the rethrow should be removed to better match how the real sync context works (at least for wpf apps), but I can probably fix that when merging if you don't have the time.